src/nupic/algorithms/SpatialPooler.cpp

```diff
@@ -134,15 +135,24 @@ UInt SpatialPooler::getNumInputs() const { return numInputs_; }
 UInt SpatialPooler::getPotentialRadius() const { return potentialRadius_; }
 
 void SpatialPooler::setPotentialRadius(UInt potentialRadius) {
-  NTA_CHECK(potentialRadius < numInputs_);
+  NTA_CHECK(potentialRadius < numInputs_/2);
```

Comment:
Instead of throwing error, this should truncate the radius to the number of inputs like the original constructor did: `potentialRadius > numInputs_ ? numInputs_ : potentialRadius`

Also, I think that the `/2` is a mistake.

Reply:
> Instead of throwing error, this should truncate the radius 

I was thinking about this and I'd prefer not to (truncate). The user asks directly the API method (set*, constructor) to do something, and if the method cannot comply, I think we should let the user know ASAP, instead of silently modifying. 

An example usecase is parameter optimization/swarming, where you'd waste cpu cycles evaluating range of values, while this might throw much earlier.  
Of course, the negative effect is that many tests need to be updated to correct values. 

> Also, I think that the /2 is a mistake. 

I think not. It's a radius, not diameter: 

> This parameter defines a square (or hyper square) area: a
column will have a max square potential pool with sides of
length (2 * potentialRadius + 1).

ok, on the other hand: 

> * If the topology is one dimensional, and the potentialRadius is 5, this
method will return an array containing 5 consecutive values centered on
the index of the column (wrapping around if necessary).
* If the topology is two dimensional (not implemented), and the
potentialRadius is 5, the method should return an array containing 25
'1's, where the exact indices are to be determined by the mapping from
1-D index to 2-D position. 

both from SpatialPooler.hpp, so either doc is wrong.